### DISCUSSION TITLE
Windows running uv_pipe_open( &pipe, 1 ); will get UV__EISDIR (-4068) error

### DISCUSSION THREAD

**Discussion by q962** [libuv/libuv#4587] [Q&A] [answered]
v1.48.0

```
		int ret  = 0;
		uv_pipe_t pipe = {};

		ret = uv_pipe_init( loop, &pipe, 0 );
		printf( "ret : %d\n", ret );

		ret = uv_pipe_open( &pipe, 1 );
		printf( "ret : %d\n", ret );

		uv_os_fd_t fd = 0;
		ret           = uv_fileno( ( void* )&pipe, &fd );
		printf( "ret : %d, fd: %d\n", ret, fd );
}
```
When using output redirection.

It works on linux, but not on windows.
Is this not supported?

**Comment by bnoordhuis** [comment]
Not supported, no. That it works on Linux is more of an accident; on Windows you need to provide an actual named pipe.

**Reply by q962** [reply]
So I can't use uv_write to write data to pipe (stdout)?

**Reply by q962** [reply]
Perhaps uv_fs_write(STDOUT_FILENO) should be used

**Reply by saghul** [reply]
What is stdout exactly in your use case? You can open it using uv_tty_t for example too.

**Reply by q962** [reply]
> When using output redirection.

It is a file

**Reply by bnoordhuis** [reply, marked as answer]
In other words, not a named pipe. In _other_ other words, not suitable for uv_pipe_open; use uv_fs_write instead.

uv_guess_handle may come in handy, too.
